src/main/java/org/squiddev/cobalt/LuaTable.java

```diff
@@ -339,20 +339,53 @@ public LuaValue getn() {
 
 	public int length() {
 		int a = getArrayLength();
-		int n = a + 1, m = 0;
-		while (!rawget(n).isNil()) {
-			m = n;
-			n += a + getHashLength() + 1;
-		}
-		while (n > m + 1) {
-			int k = (n + m) / 2;
-			if (!rawget(k).isNil()) {
-				m = k;
-			} else {
-				n = k;
+		/**
+		 * Array cannot contain nil value, except if that array is statically allocated
+		 * So if the last element is nil it means we need to binary search the array to find
+		 * the first element non nil followed with a nil value
+		 */
+		if (a > 0 && rawget(a).isNil()) {
+			int n = a+1, m = 0;
+			while (n - m > 1) {
+				int k = (m + n) / 2;
+
+				if (rawget(k).isNil() && rawget(k+1).isNil()) {
```

Comment:
AFACI, PUC Lua just does `rawget(k).isNil()`. Do we need both checks here?

Reply:
No you are right it isn't required, I did put it because of the other bug in #61 but I will remove it